packages/protocol/contracts/dapis/Api3Adminship.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "./interfaces/IApi3Adminship.sol";
+
+/// @title Adminship contract controlled by the API3 DAO
+contract Api3Adminship is IApi3Adminship {
+  string internal constant ERROR_UNAUTHORIZED = "Unauthorized";
+  string internal constant ERROR_ZERO_ADDRESS = "Zero address";
+  string internal constant ERROR_EXPIRATION_NOT_EXTENDED = "Expiration not extended";
+
+  /// @dev Meta admin sets the admin statuses of addresses and has super
+  /// admin privileges
+  address public metaAdmin;
+  mapping(address => AdminStatus) public adminStatuses;
+
+  /// @dev Reverts if the caller is not the meta admin or has admin/superadmin priviliges
+  modifier onlyAdminOrMetaAdmin() {
+    require(adminStatuses[msg.sender] >= AdminStatus.Admin || msg.sender == metaAdmin, ERROR_UNAUTHORIZED);
+    _;
+  }
+
+  /// @param _metaAdmin Address that will be set as the meta admin
+  constructor(address _metaAdmin) {
+    require(_metaAdmin != address(0), ERROR_ZERO_ADDRESS);
+    metaAdmin = _metaAdmin;
+  }
+
+  /// @notice Called by the meta admin to set the meta admin
+  /// @param _metaAdmin Address that will be set as the meta admin
+  function setMetaAdmin(address _metaAdmin) external override {
+    require(msg.sender == metaAdmin, ERROR_UNAUTHORIZED);
+    require(_metaAdmin != address(0), ERROR_ZERO_ADDRESS);
+    metaAdmin = _metaAdmin;
+    emit SetMetaAdmin(metaAdmin);
+  }
+
+  /// @notice Called by the meta admin to set the admin status of an address
+  /// @param admin Address whose admin status will be set
+  /// @param status Admin status
+  function setAdminStatus(address admin, AdminStatus status) external override {
+    require(msg.sender == metaAdmin, ERROR_UNAUTHORIZED);
+    adminStatuses[admin] = status;
+    emit SetAdminStatus(admin, status);
+  }
+
+  /// @notice Called by an admin to renounce their admin status
+  /// @dev To minimize the number of transactions the meta admin will have
+  /// to make, the contract is implemented optimistically, i.e., the admins
+  /// are expected to renounce their admin status when they are needed to.
+  /// If this is not the case, the meta admin can always revoke their
+  /// adminship.
+  /// This method cannot be used by the meta admin to renounce their meta
+  /// adminship.
+  function renounceAdminStatus() external override {
```

Comment:
Do you have any idea when this would be called? What's the use case? Maybe a better question for @bbenligiray

Reply:
Maybe it's just following a similar pattern that openzeppelin uses for the [Ownable.sol](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/406c83649bd6169fc1b578e08506d78f0873b276/contracts/access/Ownable.sol#L54) contract? I'm just guessing here

Reply:
Ah, okay. Thanks for the explanation. That makes sense. I still don't really get when it would be used, but I trust openzepplin has a good justification.

Reply:
The justification is explained here
https://github.com/api3dao/airnode/blob/8658431d252f7442906af61f1e8c6ec27d1ec134/packages/protocol/contracts/authorizers/Api3Authorizer.sol#L46
I'm more thinking about the day where the DAO assigns adminship to a multisig controlled by a team. Once that multisig becomes obsolete, we don't want to require a proposal from the DAO to pass, it would be nice if the multisig could just renounce its adminship.